app/api/events.py

```diff
@@ -577,11 +578,21 @@ def before_update_object(self, event, data, view_kwargs):
             else:
                 event.deleted_at = data.get('deleted_at')
 
+        deleted_tickets = 0
+        for ticket in event.tickets:
+            if ticket.deleted_at:
+                deleted_tickets += 1
+
         if 'is_event_online' not in data and event.is_event_online \
                 or 'is_event_online' in data and not data['is_event_online']:
-            if data.get('state', None) == 'published' and not data.get('location_name', None):
-                raise ConflictException({'pointer': '/data/attributes/location-name'},
-                                        "Location is required to publish the event")
+            if data.get('state', None) == 'published':
+                if not data.get('location_name', None):
+                    raise ConflictException({'pointer': '/data/attributes/location-name'},
+                                            "Location is required to publish the event")
+                if deleted_tickets == len(event.tickets):
```

Comment:
How does this check if there are no tickets in the event?

Makes no sense to me

Reply:
Because when there are no tickets associated with an event, the event has an array of tickets with size 0, while the `deleted_tickets` are itself zero as well. So even when event has no tickets, `event.tickets` does not throw an error, and `len(event.tickets)` returns 0.

Reply:
`event.ticket.query(deleted_at=None).count() > 1`

This is not actual code. But this is how it should be done

If there are 2 active tickets and a million deleted tickets, you'll have 1 million python objects in memory. Whereas there need to be 0

Reply:
@iamareebjamal Will make the required changes.